pandas/io/xml.py

```diff
@@ -1127,7 +1127,7 @@ def read_xml(
     """
     check_dtype_backend(dtype_backend)
 
-    if not any(
+    if isinstance(path_or_buffer, str) and not any(
```

Comment:
@mroeschke the documentation discrepancy is fixed here. Also, unit tests are passing and CI tests are green. 

Not sure how you feel about moving the warning logic to `_parse`. This isn't something that we did in similar PRs. Not sure it matters? 

Otherwise, I think this PR is ready for inspection.

Reply:
Agreed this warning would be better suited in `_parse` as well.